src/main/java/picard/analysis/GtfToRefflat.java

```diff
@@ -0,0 +1,266 @@
+    private static final String COORDINATE_DELIMITER = ",";
+    private static final String NEW_LINE_DELIMITER = "\n";
+    private static final String ATTRIBUTE_DELIMITER = " ";
+
+    private File refflat = null;
+
+    private List<String> rows = new ArrayList<>();
+    private String transcriptId = "";
+    private List<Integer> exonStarts = new ArrayList<>();
+    private List<Integer> exonEnds = new ArrayList<>();
+    private int codingStart = -1;
+    private int codingEnd = -1;
+    private String gene_id = "";
+    private String chromosome = "";
+
+    private Strand strand = null;
+    private String type = "";
+
+    public GtfToRefflat(File GTF) {
+        this.GTF = GTF;
+
+        this.doWork();
+    }
+
+
+    protected int doWork() {
+        if (GTF != null) {
+
+            // convert the Gtf to a Gff3 and create a Gff3Feature
+            final File GFF3 = convertToGFF3(GTF);
+            final AbstractFeatureReader<Gff3Feature, LineIterator> reader = AbstractFeatureReader.getFeatureReader(GFF3.getAbsolutePath(), null, new Gff3Codec(), false);
+
+            String currentTranscriptId = "";
+
+            try {
+
+                boolean hasExon = false;
+                String ignore_id = "";
+                boolean has_stopCodon = false;
+
+                // iterate through each line in the Gff3Feature
+                for (final Gff3Feature feature : reader.iterator()) {
+
+                    // if the line has no transcipt_id, move onto the next line
+                    if(!feature.getAttribute("transcript_id").isEmpty()) {
+                        currentTranscriptId = feature.getAttribute("transcript_id").get(0);
+                    } else {
+                        continue;
+                    }
+
+                    // Since information is grouped by transcipt_id, once the transcript_id is different
+                    // create a line in the refflat file with the collected information
+                    if (!transcriptId.equals(currentTranscriptId) && !transcriptId.equals("")
+                            && !ignore_id.equals(transcriptId)) {
+                        this.addRow();
+                    } else if (strand != feature.getStrand() && strand != null) {
+                        System.out.println("Error: all group members must be on the same strand");
+                        ignore_id = currentTranscriptId;
+                        resetVariables();
+                        continue;
+                    }
+
+                    gene_id = feature.getID();
+                    chromosome = feature.getContig();
+                    strand = feature.getStrand();
+                    type = feature.getType().toLowerCase();
+
+                    if (type.equals("exon")) {
+                        exonStarts.add(feature.getStart() - 1);
+                        exonEnds.add(feature.getEnd());
+                        hasExon = true;
+                    } else if (hasExon == false) {
```

Comment:
Can CDS be explicit?

Reply:
@kishorikonwar I have inherited this PR from Sophie and don't know the domain that well.  Can you answer your own question here?  If you're not sure, I'm inclined to leave the code as is.  Thanks.

Reply:
@gbggrant This is fine now and we have resolved it already, therefore, you can safely ignore it.